DP Mystery Map/Assets/Scripts/PlayerController.cs

```diff
@@ -208,13 +222,83 @@ private void GridMoveUpdate()
                 Player.FacingDirection = _walkDirection;
             }
 
+            //Update the idleFacing position per direction
+            switch (Player.FacingDirection)
```

Comment:
Put this in a new `DirectionChangeEventHandler` and subscribe it to the `Player.directionChangeEvent` event. Make sure to unsubscribe the event in the `PlayerController.OnDestroy()` method too.